tests/nvidiagpu/deploy-gpu-test.go

```diff
@@ -204,22 +261,68 @@ var _ = Describe("GPU", Ordered, Label(tsparams.LabelSuite), func() {
 
 				if nfdPkgManifestBuilderByCatalog == nil {
-					Skip("NFD packagemanifest not found in catalogsource")
-				}
+					glog.V(gpuparams.GpuLogLevel).Infof("NFD packagemanifest was not found in the default '%s'"+
+						" catalog.", nfdCatalogSourceDefault)
+
+					if createNFDCustomCatalogsource {
+						glog.V(gpuparams.GpuLogLevel).Infof("Creating custom catalogsource '%s' for NFD "+
+							"catalog.", nfdCustomCatalogSource)
+						glog.V(gpuparams.GpuLogLevel).Infof("Creating custom catalogsource '%s' for NFD "+
+							"Operator with index image '%s'", nfdCustomCatalogSource, nfdCustomCatalogsourceIndexImage)
+
+						nfdCustomCatalogSourceBuilder := olm.NewCatalogSourceBuilderWithIndexImage(inittools.APIClient,
+							nfdCustomCatalogSource, nfdCatalogSourceNamespace, nfdCustomCatalogsourceIndexImage,
+							nfdCustomCatalogSourceDisplayName, nfdCustomNFDCatalogSourcePublisherName)
+
+						Expect(nfdCustomCatalogSourceBuilder).ToNot(BeNil(), "error creating custom "+
+							"NFD catalogsource %s:  %v", nfdPackage, nfdCustomCatalogSource, err)
+
+						createdNFDCustomCatalogSourceBuilder, err := nfdCustomCatalogSourceBuilder.Create()
+						Expect(err).ToNot(HaveOccurred(), "error creating custom NFD "+
+							"catalogsource '%s':  %v", nfdPackage, nfdCustomCatalogSource, err)
+
+						Expect(createdNFDCustomCatalogSourceBuilder).ToNot(BeNil(), "Failed to "+
+							" create custom NFD catalogsource '%s'", nfdCustomCatalogSource)
+
+						By("Sleep for 30 seconds to allow the NFD custom catalogsource to be created")
+						time.Sleep(30 * time.Second)
+
+						glog.V(gpuparams.GpuLogLevel).Infof("Wait up to 4 mins for custom NFD catalogsource "+
+							"'%s' to be ready", createdNFDCustomCatalogSourceBuilder.Definition.Name)
+
+						Expect(createdNFDCustomCatalogSourceBuilder.IsReady(4 * time.Minute)).NotTo(BeFalse())
+
+						nfdPkgManifestBuilderByCustomCatalog, err := olm.PullPackageManifestByCatalog(inittools.APIClient,
+							nfdPackage, nfdCatalogSourceNamespace, nfdCustomCatalogSource)
+
+						Expect(err).ToNot(HaveOccurred(), "error getting NFD packagemanifest '%s' "+
+							"from custom catalog '%s':  %v", nfdPackage, nfdCustomCatalogSource, err)
 
-				glog.V(gpuparams.GpuLogLevel).Infof("The nfd packagemanifest name returned: %s",
-					nfdPkgManifestBuilderByCatalog.Object.Name)
+						nfdCatalogSource = nfdCustomCatalogSource
+						nfdChannel := nfdPkgManifestBuilderByCustomCatalog.Object.Status.DefaultChannel
+						glog.V(gpuparams.GpuLogLevel).Infof("NFD channel '%s' retrieved from packagemanifest "+
+							"of custom catalogsource '%s'", nfdChannel, nfdCustomCatalogSource)
 
-				nfdChannel := nfdPkgManifestBuilderByCatalog.Object.Status.DefaultChannel
-				glog.V(gpuparams.GpuLogLevel).Infof("The NFD channel retrieved from packagemanifest is:  %v",
-					nfdChannel)
+					} else {
+						Skip("NFD packagemanifest not found in default 'redhat-operators' catalogsource, " +
```

Comment:
Don't we want to fail at this point? NFD isn't found in the default catalog, and no alternative catalog is set, so we can't proceed without NFD.